internal/core/processor/universal/processor_test.go

```diff
@@ -167,121 +167,133 @@ func TestGetLastBlock(t *testing.T) {
 	}
 }
 
-func TestVerifyChainConsistency(t *testing.T) {
+func TestProcessUniversalBlockRange(t *testing.T) {
 	t.Parallel()
+
+	ctx := context.TODO()
 	tests := []struct {
-		name                   string
-		startingBlock          uint64
-		lastBlockDB            model.Block
-		lastBlockDBError       error
-		previousBlockData      *types.Header
-		previousBlockDataError error
-		expectedError          error
+		name                       string
+		startingBlock              uint64
+		previousBlockHeader        *types.Header
+		previousBlockData          model.Block
+		blockHeader                *types.Header
```

Comment:
I think BlockHeader and BlockData probably both contain the same information. Can you check if both of them are needed?

Reply:
yes, they are needed but the name is confusing. I will change it